### ISSUE TITLE
Cannot read Thai language in Edit but yes in Browser

### ISSUE THREAD

**Issue by TomChat2011** [jendrikseipp/rednotebook#375]
After installation of RedNotebook 2.3 on Windows7 I am not able to read or enter Thai language text in Edit Page which was working fine in earlier versions. Can read OK only in the Preview but that is annoying as I hardly use the Preview. 
![rnb cannot read thai](https://user-images.githubusercontent.com/32998299/31860879-06569864-b74c-11e7-8976-633768721df6.JPG)

**Comment by jendrikseipp** [comment]
Thanks for the report. The issue comes from the underlying GTK library. Could you please install the following installer https://sourceforge.net/projects/pygobjectwin32/files/pygi-aio-3.18.2_rev12-setup_549872deadabb77a91efbc56c50fe15f969e5681.exe/download, run the included gedit editor and see if the Thai characters show up or not? This is the installer that RedNotebook version 2.3 uses.

Afterwards, it would be good if you could do the same for the newer installer https://sourceforge.net/projects/pygobjectwin32/files/pygi-aio-3.24.1_rev1-setup_049a323fe25432b10f7e9f543b74598d4be74a39.exe/download.

**Comment by TomChat2011** [comment]
Thanks a lot for your prompt reply to my post. I forgot to mention that I use the portable .paf installers on both PC running Win7 and laptop Win10. Had used version 1.14 for longtime until ver 2.3 came out early this month. So I tried it on this PC running on 64 bit Win7.  Still using Ver 1.5 on Win10. Any solution you might suggest for PAF installers?

**Comment by jendrikseipp** [comment]
I'd still recommend the steps from my last response, since the .paf installer only changes how software is installed, not how it's run.

**Comment by TomChat2011** [comment]
I am not tekkie enough here. Tried to run the 1st installer and reached this point:

PyGI 3.18.2 AIO Setup

Setup cant find supported Python.
Do you want to specify its location? else installation will be cancelled.
With options for No and Yes

How do I guide the installer to find Python? I tried guiding it to where I installed my portable programmes and pointed it to RNB but still it can't find it. Do I have one?

**Comment by jendrikseipp** [comment]
Ah, you'll have to install Python manually. Please use the version that RedNotebook uses: https://www.python.org/ftp/python/3.4.4/python-3.4.4.msi

If you uses the suggested default location, the AIO installer should find your Python installation automatically.

**Comment by TomChat2011** [comment]
I guided the PyGI setup to where the Python was installed and reach the point to select these in the picture shown here. What are the item(s) I should select from this list? Thanks again.

**Comment by jendrikseipp** [comment]
Please select the following packages on the respective pages:
  1) GSpell, Webkit2GTK (not WebkitGTK), GTK+, Pango
  2) GtkSpell
  3) GIR

Not sure which page it's on, but please also select "gedit" if there's an option for it.

**Comment by TomChat2011** [comment]
I am on this page:
![rnb pygi setup](https://user-images.githubusercontent.com/32998299/31863426-bcebc376-b777-11e7-9884-b24061ca1c96.JPG)

**Comment by jendrikseipp** [comment]
OK, this is page 1, there will be 3 pages with selection options.

Select the following packages on the respective pages:
  1) GSpell, Webkit2GTK (not WebkitGTK), GTK+, Pango and Gedit (if it's there)
  2) GtkSpell
  3) GIR

**Comment by TomChat2011** [comment]
Restarted the PC after installation and tried the RNB, but still Thai cannot be read. 

I have installed the PyGI 3.18.2 AIO and carefully selected the following packages on the respective pages:
1.GSpell, Webkit2GTK (not WebkitGTK), GTK+, Pango and Gedit (if it's there)
2.GtkSpell
3.GIR

Gedit wasn't found. However I think I didn't remove the tick marked for the Base pakages on page 1. Any consequence for this?

What should be the next step? What if I try to install the newer version, will it overwrite the previous version, then maybe to select the mentioned packages again?  Or should I also reinstall the RNB 2.3 from the PAF file?

**Comment by jendrikseipp** [comment]
Have you tried a computer-wide search for "gedit"? Please don't try the newer installer for now. First, we must reproduce hte problem. If the search doesn't find "gedit", please run the installer again and if necessary just install everything.

**Comment by TomChat2011** [comment]
Thanks Jendrik, for your assistance with these all. I was away from Bangkok for a few days; now back to my PC to work on this. I found this file "Gedit.py" at location:
MyProgms\RedNotebookPortable\Python344\Lib\site-packages\gi\overrides (this is where I keep my portable programs). I installed Python in folder Python344.

Is this the Gedit that you mentioned? What should I do with it next?

**Comment by jendrikseipp** [comment]
I checked the installer again and it seems that Gedit is not included after all, only its Python bindings. So we can forget about that :-)

You could try to install the Gedit program directly: http://ftp.gnome.org/pub/GNOME/binaries/win64/gedit/gedit-x86_64-3.20.1.msi
Does it display the Thai characters correctly?

**Comment by TomChat2011** [comment]
You mean to install into the portable RNB folder? That was where I installed the Python.

I found another issue today with RNB 2.3, some long htm links are not contained in the text wrap. It tends stretch out in one long line.

**Comment by jendrikseipp** [comment]
The Gedit installer has nothing to do with the other installers. You can just click the link, install it, run it and enter some Thai characters. 

http://ftp.gnome.org/pub/GNOME/binaries/win64/gedit/gedit-x86_64-3.20.1.msi

Are the Thai characters shown correctly?

**Comment by TomChat2011** [comment]
I installed the Gedit Text Editor -- still cannot enter or read Thai text in the Edit mode of the RNB 2.3. Thai text can be read only in the preview mode. What I have not yet tried is running the Python again. Should I do that?

I still have the old version RNB 1.5 portable on this PC. There is no problem here at all with Thai text. 

When I copy those unreadable Thai characters from the RBN and paste into the Open Office text as shown, I would get the correct reading in Thai.
![rnb thai characters](https://user-images.githubusercontent.com/32998299/32144846-7d8d4dba-bcf1-11e7-80c3-d55ab2d5a1fe.JPG)

**Comment by jendrikseipp** [comment]
The question is whether Gedit Text Editor can display Thai text. This has nothing to do with RedNotebook.

**Comment by TomChat2011** [comment]
Gedit Text Editor can display unreadable Thai text, same type of characters in square-like box as appeared in the RNB. Is there somewhere in this programme that I can tweak to get the font?

**Comment by jendrikseipp** [comment]
You should be able to change fonts in Gedit. This is also possible in RedNotebook (in the options dialog), so maybe you can find a font that displays Thai characters. If you do so, please tell me which font works.

If you would like to have the display issue solved, I recommend reporting the bug to the Gedit developers: https://wiki.gnome.org/Apps/Gedit/ReportingBugs

Closing this issue since it stems from the GTK library and not from RedNotebook itself.

**Comment by TomChat2011** [comment]
Hi Jendrik,

Is this the correct method to contact you via email as the issue #375 was
closed.

Still had not solved the font issue to get the Thai language working in the
Edit mode. I checked the old versions 1.5 and 1.14 which read Thai well and
can type in as well. Can't find what font was used in the 1.5 version as
these were not available in the Preferences; while the 1.14 version's
Preferences which looked much alike the 2.3 version is using Sans font for
Edit mode, and reads Thai. The 2.3 version is also using Sans font for Edit
mode.

I may revert back to continue using 1.14 version which opens up fast and
quick to search, very instant; the 1.5 version too. The 2.3 version is
still much slower in comparison. When I cancelled the search in the 2.3
version, it would take me to a different page, sometime back to the year
2011 date. The old versions never behaved like this.

I really like the face of the 2.3 and doing away with clouds, which I
rarely used.

Best regards,
Suthi..
